src/adagrams.js

```diff
@@ -1,15 +1,175 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+class Adagrams {
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+  static drawLetters = () => {
+    // Implement this method for wave 1
+    const LETTER_POOL = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1,
+    };
+    const letters = [];
+    for (const key in LETTER_POOL) {
+      for (let i = 0; i < LETTER_POOL[key]; i++) {
+        letters.push(key);
+      }
+    }
 
-export const highestScoreFrom = (words) => {
-  // Implement this method for wave 4
-};
+    const shuffled = letters.sort(() => 0.5 - Math.random());
+    let userHand = shuffled.slice(0, 10);
+    // console.log(userHand);
+    return userHand;
+  };
+
+
+  static usesAvailableLetters = (input, lettersInHand) => {
+    // Implement this method for wave 2
+    const itemCounter = (array, item) => {
+      let counter = 0;
+      array.flat(Infinity).forEach((x) => {
```

Comment:
I'm not clear on what the purpose of the `flat(Infinity)` is here; `lettersInHand` is not a nested array, and the array of letters from the input word generated by the `split` on line 65 is not either.